services/cognito/app/operations/segments/age.rb

```diff
@@ -0,0 +1,61 @@
+# frozen_string_literal: true
+
+module Segments
+  class Age < Ros::ActivityBase
+    step :init
+    step :verify_segment
+    failed :incorrect_segment, Output(:success) => End(:failure)
+    step :translate_segment
+    step :apply_segment
+    failed :segment_not_applied, Output(:success) => End(:failure)
+
+    private
+
+    def init(ctx, segment:, **)
+      ctx[:base_date] = Time.zone.today
+      ctx[:segment] = Array.wrap(segment)
+      ctx[:translated_segments] = []
+    end
+
+    def verify_segment(_ctx, segment:, **)
+      segment.each do |element|
+        return false unless verify_segment_element(element)
+      end
+    end
+
+    def incorrect_segment(_ctx, errors:, segment:, **)
+      errors.add(:segment, "Incorrect age segment value: #{segment}")
+    end
+
+    def translate_segment(_ctx, segment:, base_date:, translated_segments:, **)
+      segment.each do |element|
+        case element
+        when Hash
+          el = element.with_indifferent_access
+          translated_segments << ((base_date - el['to'].to_i.years - 1.year + 1.day)..(base_date - el['from'].to_i.years))
```

Comment:
I think this is worth to move to a date range class that is able to build these. it looks very convoluted code